src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -133,7 +133,7 @@ private void RunGetRequestedDataAlwaysOffSampler(Activity activity)
         private void RunGetRequestedDataOtherSampler(Activity activity)
         {
             ActivityContext parentContext;
-            if (string.IsNullOrEmpty(activity.ParentId))
+            if (string.IsNullOrEmpty(activity.ParentId) || activity.ParentSpanId.ToHexString().Equals("0000000000000000"))
```

Comment:
this maynot be a perf issue, if ToHexString() is not actually allocating a string, but returns the caches string value. to be confirmed.